pinecone/index_connection_test.go

```diff
@@ -991,6 +991,35 @@ func TestToUsageUnit(t *testing.T) {
 	}
 }
 
+func TestNormalizeHost(t *testing.T) {
+	tests := []struct {
+		name         string
+		host         string
+		expectedHost string
+	}{
+		{
+			name:         "http:// scheme should be removed",
+			host:         "http://this-is-my-host.io",
+			expectedHost: "this-is-my-host.io:443",
```

Comment:
This is a bit odd... I would expect a `http://` transport to default to port 80.

However, I also don't see a user specifying http plaintext without a port for an index.